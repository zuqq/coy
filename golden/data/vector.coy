struct Vector(f64, f64, f64);

fn dot_product(v: Vector, w: Vector) -> f64 {
    let Vector(vx, vy, vz) = v;
    let Vector(wx, wy, wz) = w;
    vx * wx + vy * wy + vz * wz
}

fn main() -> () {
    let v = Vector(1.0, 0.0, 0.0);
    let w = Vector(0.0, 2.0, 0.0);
    println!("v dot w == {}", dot_product(v, w));
    println!("w dot w == {}", dot_product(w, w))
}
