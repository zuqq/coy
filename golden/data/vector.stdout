dot_product(v, w) == 0.000000
dot_product(w, w) == 4.000000
